File: trunk/OscarDotNet/component/biz/Class_biz_data_conditions.pas
unit Class_biz_data_conditions;

interface

type
  TClass_biz_data_conditions = class
  private
  public
    constructor Create;
    function BeMemberTrainee(member_id: string): boolean;
  end;

implementation

uses
  Class_biz_members;

constructor TClass_biz_data_conditions.Create;
begin
  inherited Create;
  // TODO: Add any constructor code here
end;

function TClass_biz_data_conditions.BeMemberTrainee(member_id: string): boolean;
begin
  BeMemberTrainee := (TClass_biz_members.Create.MedicalReleaseLevelOfMemberId(member_id) = 'Trainee');
end;

end.

File: trunk/OscarDotNet/component/biz/Class_biz_data_conditions.cs
using System.Collections;

namespace Class_biz_data_conditions
  {

  public class TClass_biz_data_conditions
    {

    public TClass_biz_data_conditions() : base()
      {
      }

    public bool BeMemberTrainee(string member_id)
      {
      return (new ArrayList() {"Student","BLS Intern"}).Contains(new Class_biz_members.TClass_biz_members().MedicalReleaseLevelOfMemberId(member_id));
      }

    public bool BeRolePersonnelRelated(string role_name)
      {
      return (new ArrayList() {"Squad Commander","Squad Manager","Squad Membership Coordinator","Squad Training Officer"}).Contains(role_name);
      }

    } // end TClass_biz_data_conditions

  }